classes/class-wc-connect-taxjar-integration.php

```diff
@@ -729,6 +664,84 @@ public function cleanup_tax_label( $rate_name ) {
 		return $clean_label;
 	}
 
+	/**
+	 * Replace WooCommerce tax locations with TaxJar jurisdictions.
+	 *
+	 * @param  array       $location  Location address.
+	 * @param  string      $tax_class Tax class.
+	 * @param  WC_Customer $customer  Customer.
+	 *
+	 * @return array
+	 */
+	public function get_tax_location( $location, $tax_class, $customer ) {
+		$jurisdictions = $this->get_jurisdictions( $location, $customer );
+		$county        = $jurisdictions->county ?? '';
+		$city          = $jurisdictions->city ?? '';
+
+		$location[2] = '';
+		$location[3] = trim( $county . ' ' . $city );
+
+		return $location;
+	}
+
+	/**
+	 * Replace WooCommerce tax locations for Order with TaxJar jurisdictions.
+	 *
+	 * @param  array $location  Location address.
+	 *
+	 * @return array
+	 */
+	public function get_tax_location_for_order( $location ) {
+
+		if ( ! is_null( WC()->customer ) ) {
+			$jurisdictions        = $this->get_jurisdictions( $location, WC()->customer );
+			$county               = $jurisdictions->county ?? '';
+			$city                 = $jurisdictions->city ?? '';
+			$location['postcode'] = '';
+			$location['city']     = trim( $county . ' ' . $city );
+		}
+
+		return $location;
+	}
+
+	/**
+	 * Get jurisdictions stored in transient.
+	 *
+	 * When OK response is received from TaxJar in smartcalcs_cache_request() method jurisdictions are stored in transient.
+	 *
+	 * @param  array       $location  Location address.
+	 * @param  WC_Customer $customer  Customer.
+	 *
+	 * @return object|false
+	 */
+	private function get_jurisdictions( $location, $customer ) {
+		if ( is_null( $customer ) ) {
+			return false;
+		}
+
+		$key           = 'tj_jurisdictions_' . md5( strtoupper( implode( '', $location ) . $customer->get_shipping_address() ) );
+		$jurisdictions = get_transient( $key );
+
+		return $jurisdictions;
+	}
+
+	/**
+	 * Get normalized address from request body.
+	 *
+	 * @param array $request Request.
+	 *
+	 * @return array
+	 */
+	private function get_normalized_location( $request ) {
+		return array(
+			'to_country' => strtoupper( (string) $request['to_country'] ) ?? '',
```

Comment:
```suggestion
			'to_country' => isset($request['to_country']) ? strtoupper((string) $request['to_country']) : '',
```
The ?? '' will never execute because `strtoupper()` always returns a string (even empty string), never null.